public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/PagerDutyFields/PagerDutyFields.tsx

```diff
@@ -1,18 +1,31 @@
 import React, { FC } from 'react';
-import { TextInputField, validators } from '@percona/platform-core';
+import { TextInputField } from '@percona/platform-core';
 import { Messages } from '../AddNotificationChannelModal.messages';
 import { PagerDutyFieldsProps } from './PagerDutyFields.types';
 
-export const PagerDutyFields: FC<PagerDutyFieldsProps> = ({ values }) => {
-  const { required } = validators;
-  const { routing, service } = values;
-  const routingValidators = service ? [] : [required];
-  const serviceValidators = routing ? [] : [required];
+/**
+ * This is a validator factory.
+ * Given a key, return a validator that will check if there's anything there.
+ */
+const anotherKeyRequired = (complementaryKey: string) => (value: any, allValues: any): undefined | string => {
```

Comment:
We have a type called `Validator` (core I think), you may want to use it here.

Reply:
Right, just changed it